src/test/java/edu/harvard/iq/dataverse/engine/command/impl/RestrictFileCommandTest.java

```diff
@@ -91,6 +92,7 @@ public void testRestrictUnpublishedFile() throws CommandException{
     public void testRestrictPublishedFile() throws Exception{
         file.setOwner(dataset);
         dataset.setPublicationDate(new Timestamp(new Date().getTime()));
+        file.setPublicationDate(dataset.getPublicationDate());
```

Comment:
This test was passing with the original code, but I don't think it should have - the file is not released unless the date is set. What I think was really happening was that the call to dataset.getEditVersion was not finding a version that had this file in its list of metadatas, so no change was being made to the file.